lib/protobuf/active_record/serialization.rb

```diff
@@ -131,33 +131,54 @@ def protobuf_message(message = nil, options = {})
 
           @protobuf_message
         end
+
+        # :nodoc:
+        def _deprecated_fields
+          return @_deprecated_fields if @_deprecated_fields
+          @_deprecated_fields = protobuf_message.all_fields.map do |field|
+            next if field.nil?
+            next unless field.deprecated?
+            field.name.to_sym
+          end
+          @_deprecated_fields.flatten!
+          @_deprecated_fields.compact!
+          @_deprecated_fields.uniq!
+
+          @_deprecated_fields
+        end
+
+        # :nodoc:
+        def _protobuf_fields
+          fields = protobuf_message.all_fields.map do |field|
+            next if field.nil?
+            field.name.to_sym
+          end
+          fields.flatten!
+          fields.compact!
+          fields.uniq!
+
+          fields
+        end
+
+        # :nodoc:
+        def _mapped_protobuf_fields
+          return @_mapped_protobuf_fields if @_mapped_protobuf_fields
+          @_mapped_protobuf_fields = _protobuf_fields.select do |field|
+            _protobuf_field_transformers.key?(field) || instance_methods.include?(field)
```

Comment:
my PR is doing something very similar, but in this case we have to check `respond_to?` at the instance level and can't rely on `instance_methods` because AR is dynamically generating `respond_to?` responses (even if the methods are not included in `instance_methods`)